monai/losses/aucm_loss.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from typing import Callable, List, Optional, Union
+
+import torch
+from torch.nn.modules.loss import _Loss
+
+from monai.networks import one_hot
+from monai.utils import LossReduction
+
+
+class AUCMLoss(_Loss):
+    """
+    AUCM Loss with squared-hinge function: a novel loss function to directly optimize AUROC
+
+    The original paper: Large-scale Robust Deep AUC Maximization: A
+    New Surrogate Loss and Empirical Studies on Medical Image Classification
+    https://arxiv.org/abs/2012.03173
+    """
+
+    def __init__(
+        self,
+        margin: float = 1.0,
+        imratio: List[float] = None,
```

Comment:
Hi @JohnMasoner,
flake8 error show the `imratio` arg still has some type error, I would suggest you to change this arg name to `pos_weight` refer to PyTorch if I understand correctly. And also correct the type of this args refer to PyTorch.
https://github.com/pytorch/pytorch/blob/b136f3f310aa01a8b3c1e63dc0bfda8fd2234b06/torch/nn/modules/loss.py#L694-L695